specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json

```
@@ -0,0 +1,4472 @@
+          "description": "Aggregate device health state"
+        },
+        "digitalTwinModelId": {
+          "type": "string",
+          "description": "Digital twin model Id"
+        },
+        "healthChecks": {
+          "type": "array",
+          "description": "Array of health checks and their results",
+          "items": {
+            "$ref": "#/definitions/HealthCheck"
+          }
+        }
+      },
+      "required": [
+        "deviceId",
+        "state",
+        "healthChecks"
+      ]
+    },
+    "DeviceHealthState": {
+      "description": "Device health states",
+      "type": "string",
+      "enum": [
+        "healthy",
+        "unhealthy"
+      ],
+      "x-ms-enum": {
+        "name": "DeviceHealthState",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "healthy",
+            "name": "healthy",
+            "description": "Agent is healthy"
+          },
+          {
+            "value": "unhealthy",
+            "name": "unhealthy",
+            "description": "Agent is in an unhealthy state"
+          }
+        ]
+      }
+    },
+    "DeviceHealthList": {
+      "description": "Array of Device Health, with server paging support.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The collection of pageable items.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DeviceHealth"
+          }
+        },
+        "nextLink": {
+          "description": "The link to the next page of items.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "HealthCheck": {
+      "description": "Health check",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Health check name"
+        },
+        "result": {
+          "$ref": "#/definitions/HealthCheckResult",
+          "description": "Health check result"
+        }
+      }
+    },
+    "DeviceHealthFilter": {
+      "type": "object",
+      "properties": {
+        "state": {
+          "$ref": "#/definitions/DeviceHealthState",
+          "description": "Device health state"
+        },
+        "deviceId": {
+          "type": "string",
+          "description": "Device Id"
+        },
+        "moduleId": {
+          "type": "string",
+          "description": "Module Id"
+        }
+      },
+      "description": "Device health filter."
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Version of the API to be used with the client request."
+    },
+    "EndpointParameter": {
+      "name": "endpoint",
+      "in": "path",
+      "required": true,
+      "format": "url",
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "description": "Account endpoint.",
```

Device Update is part of IoT Hub which itself has its own sets of endpoints and account names. I can see this easily leading to user confusion. I would also be more precise on the desired shape (endpoint hostname, or a full URI w/ protocol):

e.g.
```suggestion
      "description": "The Device Update account endpoint (hostname only, no protocol).",
```